# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/jimmy/zgv/more-modes.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./src/modesel.c.orig	2003-04-05 17:41:04.000000000 +0300
+++ ./src/modesel.c	2003-08-09 14:06:55.000000000 +0300
@@ -28,8 +28,12 @@
   { G360x480x256,	360,480,8,1,	RK_NO_KEY,	0,'8' },
   
   /* SVGA 8-bit modes */
+  { G400x300x256,	400,300,8,1,	RK_NO_KEY,	0,RK_NO_KEY },
+  { G512x384x256,	512,384,8,1,	RK_NO_KEY,	0,RK_NO_KEY },
   { G640x480x256,	640,480,8,1,	RK_F1,		0,RK_F1 },
+  { G720x540x256,	720,540,8,1,	RK_NO_KEY,	0,RK_NO_KEY },
   { G800x600x256,	800,600,8,1,	RK_F2,		0,RK_F2 },
+  { G960x720x256,	960,720,8,1,	RK_NO_KEY,	0,RK_NO_KEY },
   { G1024x768x256,	1024,768,8,1,	RK_F3,		0,RK_F3 },
   { G1280x1024x256,	1280,1024,8,1,	RK_F4,		0,RK_F4 },
   { G1152x864x256,	1152,864,8,1,	RK_NO_KEY,	1,RK_F2 },
@@ -38,8 +42,12 @@
   /* SVGA 15-bit modes */
   { G320x200x32K,	320,200,15,2,	RK_NO_KEY,	0,RK_F5 },
   { G320x240x32K,	320,240,15,2,	RK_NO_KEY,	0,RK_NO_KEY },
+  { G400x300x32K,	400,300,15,2,	RK_NO_KEY,	0,RK_NO_KEY },
+  { G512x384x32K,	512,384,15,2,	RK_NO_KEY,	0,RK_NO_KEY },
   { G640x480x32K,	640,480,15,2,	RK_NO_KEY,	0,RK_F8 },
+  { G720x540x32K,	720,540,15,2,	RK_NO_KEY,	0,RK_NO_KEY },
   { G800x600x32K,	800,600,15,2,	RK_NO_KEY,	0,RK_SHIFT_F1 },
+  { G960x720x32K,	960,720,15,2,	RK_NO_KEY,	0,RK_NO_KEY },
   { G1024x768x32K,	1024,768,15,2,	RK_NO_KEY,	0,RK_SHIFT_F4 },
   { G1280x1024x32K,	1280,1024,15,2,	RK_NO_KEY,	0,RK_SHIFT_F7 },
   { G1152x864x32K,	1152,864,15,2,	RK_NO_KEY,	1,RK_F3 },
@@ -48,8 +56,12 @@
   /* SVGA 16-bit modes */
   { G320x200x64K,	320,200,16,2,	RK_NO_KEY,	0,RK_F6 },
   { G320x240x64K,	320,240,16,2,	RK_NO_KEY,	0,RK_NO_KEY },
+  { G400x300x64K,	400,300,16,2,	RK_NO_KEY,	0,RK_NO_KEY },
+  { G512x384x64K,	512,384,16,2,	RK_NO_KEY,	0,RK_NO_KEY },
   { G640x480x64K,	640,480,16,2,	RK_NO_KEY,	0,RK_F9 },
+  { G720x540x64K,	720,540,16,2,	RK_NO_KEY,	0,RK_NO_KEY },
   { G800x600x64K,	800,600,16,2,	RK_NO_KEY,	0,RK_SHIFT_F2 },
+  { G960x720x64K,	960,720,16,2,	RK_NO_KEY,	0,RK_NO_KEY },
   { G1024x768x64K,	1024,768,16,2,	RK_NO_KEY,	0,RK_SHIFT_F5 },
   { G1280x1024x64K,	1280,1024,16,2,	RK_NO_KEY,	0,RK_SHIFT_F8 },
   { G1152x864x64K,	1152,864,16,2,	RK_NO_KEY,	1,RK_F4 },
@@ -64,8 +76,12 @@
    */
   { G320x200x16M32,	320,200,24,4,	RK_NO_KEY,	0,RK_F7 },
   { G320x240x16M32,	320,240,24,4,	RK_NO_KEY,	0,'^' },
+  { G400x300x16M32,	400,300,24,4,	RK_NO_KEY,	0,RK_NO_KEY },
+  { G512x384x16M32,	512,384,24,4,	RK_NO_KEY,	0,RK_NO_KEY },
   { G640x480x16M32,	640,480,24,4,	RK_NO_KEY,	0,RK_F10 },
+  { G720x540x16M32,	720,540,24,4,	RK_NO_KEY,	0,RK_NO_KEY },
   { G800x600x16M32,	800,600,24,4,	RK_NO_KEY,	0,RK_SHIFT_F3 },
+  { G960x720x16M32,	960,720,24,4,	RK_NO_KEY,	0,RK_NO_KEY },
   { G1024x768x16M32,	1024,768,24,4,	RK_NO_KEY,	0,RK_SHIFT_F6 },
   { G1280x1024x16M32,	1280,1024,24,4,	RK_NO_KEY,	1,RK_F1 },
   { G1152x864x16M32,	1152,864,24,4,	RK_NO_KEY,	1,RK_F5 },
@@ -79,8 +95,12 @@
    */
   { G320x200x16M,	320,200,24,3,	RK_NO_KEY,	0,RK_F7 },
   { G320x240x16M,	320,240,24,3,	RK_NO_KEY,	0,'^' },
+  { G400x300x16M,	400,300,24,3,	RK_NO_KEY,	0,RK_NO_KEY },
+  { G512x384x16M,	512,384,24,3,	RK_NO_KEY,	0,RK_NO_KEY },
   { G640x480x16M,	640,480,24,3,	RK_NO_KEY,	0,RK_F10 },
+  { G720x540x16M,	720,540,24,3,	RK_NO_KEY,	0,RK_NO_KEY },
   { G800x600x16M,	800,600,24,3,	RK_NO_KEY,	0,RK_SHIFT_F3 },
+  { G960x720x16M,	960,720,24,3,	RK_NO_KEY,	0,RK_NO_KEY },
   { G1024x768x16M,	1024,768,24,3,	RK_NO_KEY,	0,RK_SHIFT_F6 },
   { G1280x1024x16M,	1280,1024,24,3,	RK_NO_KEY,	1,RK_F1 },
   { G1152x864x16M,	1152,864,24,3,	RK_NO_KEY,	1,RK_F5 },
